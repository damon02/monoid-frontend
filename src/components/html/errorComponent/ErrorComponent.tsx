import * as React from 'react'

interface IErrorComponentProps {
  message: string
  faIcon?: string
  onClick?: () => void
  align?: 'left' | 'center' | 'right'
}

const ErrorComponent : React.FunctionComponent<IErrorComponentProps> = (props) => (
  props.message
    ? (
      <div 
        className={`error ${props.align || 'center'}${props.onClick ? ' clickable' : ''}`}
        onClick={() => props.onClick ? props.onClick() : ({})}
      >
        {props.faIcon ? <i className={props.faIcon}/> : null}
        <div className="message">
          {props.message}
        </div>
      </div>
      )
    : null
)

export default ErrorComponent