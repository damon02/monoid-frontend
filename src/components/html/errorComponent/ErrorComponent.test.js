import React from 'react'
import ReactDOM from 'react-dom'
import ErrorComponent from './ErrorComponent'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <ErrorComponent />, 
      div
    )
  })
  
  it('renders correctly', () => {
    const tree = Renderer
      .create(<ErrorComponent message={'blahblah'} />)
      .toJSON()
  
    expect(tree).toMatchSnapshot()
  })
  
  it('changes input and then renders correctly', () => {
    const input = shallow(<ErrorComponent message={'bloehbloeh'} />)
  
    expect(input.containsMatchingElement(<div className="message">bloehbloeh</div>))  
    input.setProps({ value: 'This is a test' })
  
    expect(input.containsMatchingElement(<div className="message">This is a test</div>))
  })