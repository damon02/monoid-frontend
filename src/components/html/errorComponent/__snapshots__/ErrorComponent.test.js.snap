// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="error center"
  onClick={[Function]}
>
  <div
    className="message"
  >
    blahblah
  </div>
</div>
`;
